lilypad/server/models/functions.py

```diff
@@ -16,9 +16,8 @@
 class _FunctionBase(BaseSQLModel):
     """Base Function Model."""
 
+    project_id: int | None = Field(default=None, foreign_key=f"{PROJECT_TABLE_NAME}.id")
     name: str = Field(nullable=False, index=True)
-    hash: str = Field(nullable=False, index=True)
```

Comment:
can you explain to me why hash and code shouldn't be included in base when they are in every other model?

Reply:
because some models need optional and others dont, so if i set it on base and then override, there are type errors

Reply:
got it, makes sense thanks